Source/DafnyDriver/Commands/AuditCommand.cs

```diff
@@ -21,5 +57,9 @@ class AuditCommand : ICommandSpec {
   public void PostProcess(DafnyOptions dafnyOptions, Options options, InvocationContext context) {
     dafnyOptions.Compile = false;
     dafnyOptions.Verify = false;
+    dafnyOptions.AuditProgram = true;
+    dafnyOptions.AuditorReportFile = dafnyOptions.Get(ReportFileOption);
```

Comment:
I would prefer not having the field `DafnyOptions.AuditorReportFile` and instead using `dafnyOptions.Get(ReportFileOption)` everywhere. That way, `DafnyOptions` doesn't contain code related to many different things, and the option code can be located purely where the option is used.

Reply:
I believe that would require moving the option definition back into `DafnyCore` instead of having it be a member of the `AuditCommand` class, since it would need to be checked from `DafnyCore` code (i.e., from the auditor code). Are you happy with having the option move back?

Reply:
I would move AuditCommand and options to DafnyCore then

Reply:
I've moved the `audit` options into the `Auditor` class, so the rest of the codebase (except `AuditCommand`) doesn't need to know about them.

I'm not entirely sure I understand what goal you have in mind for the plugin mechanism here, though.

Reply:
> I'm not entirely sure I understand what goal you have in mind for the plugin mechanism here, though.

Then `Auditor/Auditor.cs`, `Auditor/AuditReport.cs` could move to DafnyDriver, and `Resolver.cs` wouldn't need changes.